tests/kernel/userbufer_validate/src/userbuffer_validate.c

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <misc/printk.h>
+#include <mmustructs.h>
+#include <arch/x86/arch.h>
+#include <ztest.h>
+
+void x86_mmu_set_pte_rw_bit(void *addr, bool value)
+{
+	u32_t pde = MMU_PDE_NUM(addr);
+	u32_t pte = MMU_PAGE_NUM(addr);
+
+	volatile struct x86_mmu_page_table *pte_addr =
+		(struct x86_mmu_page_table *)
+		(X86_MMU_PD->entry[pde].pt.page_table << 12);
+
+	pte_addr->entry[pte].rw = (u32_t)value;
+}
+
+void x86_mmu_set_pte_us_bit(void *addr, bool value)
+{
+	u32_t pde = MMU_PDE_NUM(addr);
+	u32_t pte = MMU_PAGE_NUM(addr);
+	volatile struct x86_mmu_page_table *pte_addr =
+		(struct x86_mmu_page_table *)
+		(X86_MMU_PD->entry[pde].pt.page_table << 12);
+
+	pte_addr->entry[pte].us = (u32_t)value;
+}
+
+void x86_mmu_set_pde_us_bit(void *addr, bool value)
+{
+	u32_t pde = MMU_PDE_NUM(addr);
+
+	X86_MMU_PD->entry[pde].pt.us = (volatile u32_t)value;
+
+}
+
+static int status;
+
+/* if Failure occurs
+ * _x86_mmu_buffer_validate return -EPERM
+ * else return 0.
+ * Below conditions will be tested accordingly
+ *
+ */
+
+static int buffer_rw_read(void)
+{
+	/* read write testing */
+	x86_mmu_set_pte_rw_bit((void *)0x00405000, 0);
+	status = _x86_mmu_buffer_validate((void *)0x00405000,
+			10, BUFF_WRITEABLE); /* should fail */
+	if (status != -EPERM) {
+		TC_PRINT("buffer_rw_read failed\n");
+		return TC_FAIL;
+	}
+
+	return TC_PASS;
+}
+
+static int buffer_writeable_write(void)
+{
+	x86_mmu_set_pte_rw_bit((void *)0x00405000, 1);
+	status = _x86_mmu_buffer_validate((void *)0x00405000, 10, 0x1);
+	/* should pass */
+	if (status != 0) {
+		TC_PRINT("buffer_rw_write failed\n");
+		return TC_FAIL;
+	}
+
+	return TC_PASS;
+}
+
+static int buffer_readable_read(void)
+{
+	x86_mmu_set_pte_rw_bit((void *)0x00405000, 0);
+	status = _x86_mmu_buffer_validate((void *)0x00405000, 10, 0x0);
+	/* should pass */
+	if (status != 0) {
+		TC_PRINT("buffer_readble_read failed\n");
+		return TC_FAIL;
+	}
+
+	return TC_PASS;
+}
+
+static int buffer_readable_write(void)
+{
+	x86_mmu_set_pte_rw_bit((void *)0x00405000, 1);
+	status = _x86_mmu_buffer_validate((void *)0x00405000, 10, 0x0);
+	/* should pass */
+	if (status != 0) {
+		TC_PRINT("buffer_readable_write failed\n");
+		return TC_FAIL;
+	}
+
+	return TC_PASS;
+}
+
+static int buffer_supervisor_rw(void)
+{
+	/* user/supervisor test */
+	x86_mmu_set_pte_us_bit((void *)0x00405000, 0); /* supervisor */
+	status = _x86_mmu_buffer_validate((void *)0x00405000,
+			10, 0x1 | 0x2); /*  should fail */
```

Comment:
parameters should be aligned to last '(' + 1 space, so here the 10, ... should be right belo (void *...

and the comment /* should fail */, put it on its own line.

Apply that rule everywhere

Reply:
breaking rule applied and corrected everywhere.